librz/arch/dwarf_process.c

```diff
@@ -1696,12 +1696,14 @@ static bool variable_from_die(
 	RZ_BORROW RZ_IN RZ_NONNULL DwContext *ctx,
 	RZ_BORROW RZ_IN RZ_NONNULL const RzBinDwarfDie *die) {
 	RzAnalysisDwarfVariable v = { 0 };
+	bool res = false;
 	if (!function_var_parse(ctx, NULL, NULL, &v, die, NULL)) {
-		variable_fini(&v);
-		return false;
+		goto beach;
 	}
-
-	return try_create_var_global(ctx, die, &v);
+	res = try_create_var_global(ctx, die, &v);
```

Comment:
Please note, that one of the `variable_fini()` was removed in 07a7b0d431420d9a921eeb7951271fdb332dc4f8
Sadly, I don't have that file to test on.

Reply:
I don't know how to reproduce it, but so far it looks like the memory leak is caused here, let's see what wargio has to say about it